File: CSharp/LeetCode/1800_1899/1822_SignOfTheProductOfAnArray.cs
namespace LeetCode;

class SignOfTheProductOfAnArray
{
    public string SignOfTheProductOfAnArrayLinear(int[] nums)
    {
        int minusCount = 0;

        foreach (int num in nums)
        {
            if (num == 0) return 0;
            if (num < 0) minusCount++;
        }

        return minusCount % 2 == 2 ? 1 : -1;
    }
}

File: JavaScript/1800_1899/1822_SignOfTheProductOfAnArray.js
function SignOfTheProductOfAnArrayLinear(nums) {
	let minusCount = 0;

	for (let num of nums) {
		if (num == 0) return 0;
		if (num < 0) minusCount++;
	}

	return minusCount % 2 == 0 ? 1 : -1;
}
